include/pros/rotation.h

```diff
@@ -0,0 +1,187 @@
+ *
+ * This file should not be modified by users, since it gets replaced whenever
+ * a kernel upgrade occurs.
+ *
+ * Copyright (c) 2017-2020, Purdue University ACM SIGBots.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+#ifndef _PROS_ROTATION_H_
+#define _PROS_ROTATION_H_
+
+#include <stdbool.h>
+#include <stdint.h>
+
+#ifdef __cplusplus
+extern "C" {
+namespace pros {
+namespace c {
+#endif
+
+/**
+ * Resets Rotation Sensor 
+ *
+ * Resets rotation sensor by multiplying the rotation reading by 
+ * -1 and changes the angle by -180 degrees only if the direction 
+ * was recently reversed. 
+ *
+ * This function uses the following values of errno when an error state is
+ * reached:
+ * ENXIO - The given value is not within the range of V5 ports (1-21).
+ * ENODEV - The port cannot be configured as an Rotation Sensor
+ *
+ * \param port
+ *        The V5 Rotation Sensor port number from 1-21
+ * \return 1 if the operation was successful or PROS_ERR if the operation
+ * failed, setting errno.
+ */
+int32_t rotation_reset(uint8_t port);
+
+/**
+ * Set the rotation sensor position reading to a desired rotation value.
+ *
+ * This function uses the following values of errno when an error state is
+ * reached:
+ * ENXIO - The given value is not within the range of V5 ports (1-21).
+ * ENODEV - The port cannot be configured as an Rotation Sensor
+ *
+ * \param port
+ *		  The V5 Rotation Sensor port number from 1-21
+ * \param position 
+ * 		  The position in terms of ticks
+ * \return 1 if the operation was successful or PROS_ERR if the operation
+ * failed, setting errno.
+ */
+int32_t rotation_set_position(uint8_t port, uint32_t position);
+
+/**
+ * Resets the rotation sensor position to 0.
+ *
+ * This function uses the following values of errno when an error state is
+ * reached:
+ * ENXIO - The given value is not within the range of V5 ports (1-21).
+ * ENODEV - The port cannot be configured as an Rotation Sensor
+ *
+ * \param port
+ *		  The V5 Rotation Sensor port number from 1-21
+
+ * \return 1 if the operation was successful or PROS_ERR if the operation
+ * failed, setting errno.
+ */
+int32_t rotation_reset_position(uint8_t port);
+
+/**
+ * Get the Rotation sensor's current position.  Specifically, in
+ * terms of hundreths of degrees.
+ *
+ * This function uses the following values of errno when an error state is
+ * reached:
+ * ENXIO - The given value is not within the range of V5 ports (1-21).
+ * ENODEV - The port cannot be configured as an Rotation Sensor
+ *
+ * \param  port
+ * 				 The V5 Rotation Sensor port number from 1-21
+ * \return The position value or PROS_ERR_F if the operation failed, setting
+ * errno.
+ */
+int32_t rotation_get_position(uint8_t port);
+
+/**
+ * Get the Rotation sensor's current velocity in hundreths of
+ * degrees per second.
```

Comment:
could go with centidegrees here to make it more concise, but if you don't, make sure to correct "hundreth" -> "hundredth"